### PR TITLE
Add support for specifying RGW networks

### PR DESCRIPTION
**Pull request by priteau**

### CODE REVIEW

File: roles/cephadm/templates/cluster.yml.j2 [hunk 1]
```diff
@@ -49,6 +49,9 @@ placement:
 {% if service.count_per_host is defined %}
   count_per_host: {{ service.count_per_host }}
 {% endif %}
+{% if service.networks is defined %}
```
Comment (mnasiadka) [inline roles/cephadm/templates/cluster.yml.j2:52]: let's check if it's a list

Comment (priteau) [reply roles/cephadm/templates/cluster.yml.j2:52]: Done in updated change.

File: roles/cephadm/templates/cluster.yml.j2 [hunk 2]
```diff
@@ -49,6 +49,14 @@ placement:
...
+{% if service.networks is string %}
+networks:
+  - "{{ service.networks }}"
+{% else %}
+networks: {{ service.networks }}
```
Comment (markgoddard) [inline roles/cephadm/templates/cluster.yml.j2:57]: Does this template to YAML, or do we need something like the spec below?

Comment (priteau) [reply roles/cephadm/templates/cluster.yml.j2:57]: > Does this template to YAML, or do we need something like the spec below?

It turns this in `cephadm.yml`:

```
cephadm_radosgw_services:
  - id: "rgw"
    count_per_host: 2
    networks:
      - "192.168.0.0/24"
```

into this in `cephadm_cluster.yml`:

```
service_type: rgw
service_id: rgw
placement:
  label: rgw
  count_per_host: 2
networks: ['192.168.0.0/24']
```

**Review by markgoddard — APPROVED** [review]

File: roles/cephadm/templates/cluster.yml.j2 (hunk 2 above)
Comment (markgoddard) [reply roles/cephadm/templates/cluster.yml.j2]: Would probably be fine, but I think it was printing python rather than YAML.

Comment (priteau) [reply roles/cephadm/templates/cluster.yml.j2]: I think you're right, the single quotes smell of Python.

**Pull request merged**